data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/dao/ScalarDBDao.java

```diff
@@ -0,0 +1,430 @@
+package com.scalar.db.dataloader.core.dataimport.dao;
+
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutBuilder.Buildable;
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scan;
+import com.scalar.db.api.ScanBuilder;
+import com.scalar.db.api.Scanner;
+import com.scalar.db.common.error.CoreError;
+import com.scalar.db.dataloader.core.ScanRange;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.transaction.CrudException;
+import com.scalar.db.io.Column;
+import com.scalar.db.io.Key;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The generic DAO that is used to scan ScalarDB data */
+public class ScalarDBDao {
+
+  /* Class logger */
+  private static final Logger logger = LoggerFactory.getLogger(ScalarDBDao.class);
+  private static final String GET_COMPLETED_MSG = "GET completed for %s";
+  private static final String PUT_COMPLETED_MSG = "PUT completed for %s";
+  private static final String SCAN_START_MSG = "SCAN started...";
+  private static final String SCAN_END_MSG = "SCAN completed";
+
+  /**
+   * Retrieve record from ScalarDB instance in storage mode
+   *
+   * @param namespace Namespace name
+   * @param tableName Table name
+   * @param partitionKey Partition key
+   * @param clusteringKey Optional clustering key for get
+   * @param storage Distributed storage for ScalarDB connection that is running in storage mode.
+   * @return Optional get result
+   * @throws ScalarDBDaoException if something goes wrong while reading the data
+   */
+  public Optional<Result> get(
+      String namespace,
+      String tableName,
+      Key partitionKey,
+      Key clusteringKey,
+      DistributedStorage storage)
+      throws ScalarDBDaoException {
+
+    String printKey = keysToString(partitionKey, clusteringKey);
```

Comment:
Maybe we need to add some comment here to describe "what printKey" means. Or maybe rename the variable.